### DISCUSSION TITLE
Auto update code and show diff view for custom prompt

### DISCUSSION THREAD

**Discussion by Devora** [gitbito/bitoai#75] [Q&A] [answered]
Is there a way create a custom prompt that updates the code and opens the Diff View same as in some of built in prompts?

**Comment by bitoandy** [comment, marked as answer]
Hi @Devora ,
Currently we do not have the support for opening of diff view for custom prompts. This feedback is noted and we have added the requirement for the same to our feature list for prioritization on our roadmap.

**Reply by bitoandy** [reply]
The functionality is now supported:

![Screenshot 2024-10-20 at 8 17 35 PM](https://github.com/user-attachments/assets/ff27ce0d-b496-4c3e-8d36-5a03a45f363f)

**Comment by wxwok** [comment]
By now, Bito can understand and modify multiple source code files for me by pasting them to question input. Bito splits files in file boxes in its answer. Currently, there is a copy button and a copy to editor button.  Adding a different button to compare with the origin file would be great.  It makes developers faster on merging code from Bito.

**Reply by bitoandy** [reply]
@wxwok , we are working on adding "Apply Change" functionality which will provide a way to compare the changes and accept or reject them.

**Reply by wxwok** [reply]
When asking questions, selecting files from the project browser of IDEs may be a good idea. I have to copy/paste all file contents to the input box for the question-related source code files, and it often doesn't include file names. If we provide content with a filename list and recognize it from AI's reply, it can apply changes more precisely and conveniently.
